src/common/components/file-issue-details-button.tsx

```diff
@@ -36,6 +37,12 @@ export class FileIssueDetailsButton extends React.Component<FileIssueDetailsButt
         this.state = { showingFileIssueDialog: false, showingHelpText: false };
     }
 
+    private static getDerivedStateFromProps(props: FileIssueDetailsButtonProps): FileIssueDetailsButtonState {
```

Comment:
This is the first usage of `getDerivedStateFromProps` in our codebase and it confused me a lot as a reader (it's really not apparent if you don't already know about this feature that a private static method is going to be involved in the render/update pipeline without doing any particular registration steps). It also confuses VS code (which thinks this is dead code).

Would you consider doing this with `onComponentDidUpdate` instead? I understand that doing so would trade adding an extra re-render in this path, but I think I'd much rather take that hit if it helps future maintainers to understand the code more easily.

Reply:
I'm wondering if there is a reason that you are avoiding using ```ComponentDidUpdate``` ?

Reply:
`getDerivedStateFromProps` looked like a closer match for what's happening here, and helped avoid the re-render, but I can take a look at using `componentDidUpdate` and see if that works out more familiarly.

Reply:
OK, pushed changes to do this with `componentDidUpdate`. Not much different in the main code (calls `setState` rather than returning new state, and adds an extra condition to the test to prevent looping). 

The test code is a little bit more complex in that we need to be mounted to be able to call `setState`, whereas `getDerivedStateFromProps` could be tested in isolation.